bitcoind-tests/tests/test_csfs.rs

```diff
@@ -178,16 +179,48 @@ pub fn test_desc_satisfy(cl: &ElementsD, testdata: &TestData, desc: &str) -> Vec
             let sig = secp.sign_schnorr_with_aux_rand(&msg, keypair, &aux_rand);
             Some(sig)
         }
+
+        fn lookup_price_oracle_sig(
+                &self,
+                pk: &bitcoin::XOnlyPublicKey,
+                time: u64,
+            ) -> Option<(secp256k1::schnorr::Signature, i64, u64)> {
+            let xpk = pk.to_x_only_pubkey();
+            let known_xpks = &self.0.pubdata.x_only_pks;
+            let i = known_xpks.iter().position(|&x| x == xpk).unwrap();
+
+            // select a time ahead of the current test time
+            let time_signed = time + 1;
+            let price = self.0.pubdata.price as u64;
+            let sighash_msg = sighash_msg_price_oracle_1(time_signed, price);
+            let keypair = &self.0.secretdata.x_only_keypairs[i];
+            let mut aux_rand = [0u8; 32];
+            rand::thread_rng().fill_bytes(&mut aux_rand);
+
+            let secp = secp256k1::Secp256k1::new();
+            let sig = secp.sign_schnorr_with_aux_rand(&sighash_msg, keypair, &aux_rand);
```

Comment:
In 5e533b71c8904f8ad2e0f6ec7ec50f33e3ecd13a:

Lol, very prudent of you to use aux rand.